build.gradle

```diff
@@ -524,6 +525,11 @@ subprojects {
         if (generateFilesTask != null) {
             licenseTask.dependsOn generateFilesTask
         }
+
+        def generateManifestFilesTask = project.tasks.findByName('generateComponentManifestClassFiles')
```

Comment:
This could be avoided if we have a task like:

```
task autogenerateCdkFiles(type: Exec) {
    dependsOn 'generateProtocolClassFiles'
    dependsOn 'generateComponentManifestClassFiles'
}
```

The hypothesis is that every time we will autogenerate files, we will want the license to be apply. Let me know if you think this change makes sense and should be applied as part of this PR

Reply:
Yeah it's probably fine to bundle these two together. Although I'm not entirely sure how your second statement relates to above? That does also seem fine that we want the license to apply

Reply:
My concern is that every time we add a "generateX" task, we will need to add a block of code that looks like:
```
        def generateX = project.tasks.findByName('generateX')
        if (generateX != null) {
            licenseTask.dependsOn generateX
        }
```

If we have the autogenerateCdkFiles close to the generation task, we only have to update autogenerateCdkFiles and the license addition will be applied. So there is still something to add but it's closer to where the modification is done. That being said, it add complexity on the gradle dependencies (and for a newbie in gradle like me, this seems like a big drawback)